src/main/java/io/cryostat/net/web/http/api/v2/AuthPostHandler.java

```diff
@@ -96,14 +99,32 @@ public boolean isAsync() {
 
     @Override
     public IntermediateResponse<UserInfo> handle(RequestParameters requestParams) throws Exception {
-        return new IntermediateResponse<UserInfo>()
-                .addHeader(WebServer.AUTH_SCHEME_HEADER, auth.getScheme().toString())
-                .body(
-                        auth.getUserInfo(
-                                        () ->
-                                                requestParams
-                                                        .getHeaders()
-                                                        .get(HttpHeaders.AUTHORIZATION))
-                                .get());
+
+        try {
+            Optional<IntermediateResponse<UserInfo>> redirectResponse =
+                    auth.sendLoginRedirectIfRequired(
+                            () -> requestParams.getHeaders().get(HttpHeaders.AUTHORIZATION),
+                            resourceActions());
+
+            if (!redirectResponse.isEmpty()) {
+                return redirectResponse.get();
```

Comment:
Might be nice to avoid the `if not empty, return optional.get()` pattern and instead use `Optional#orElse` or something like it, ie:

```
Optional response = redirectIfRequired().orElse(new Response()/* normal non-redirect response*/).get();
return response;
```